x/merkledb/trieview.go

```diff
@@ -541,7 +549,16 @@ func (t *trieView) GetMerkleRoot(ctx context.Context) (ids.ID, error) {
 	if err := t.calculateNodeIDs(ctx); err != nil {
 		return ids.Empty, err
 	}
-	return t.root.id, nil
+	return t.getMerkleRoot(), nil
+}
+
+func (t *trieView) getMerkleRoot() ids.ID {
+	if t.root.valueDigest.IsNothing() && len(t.root.children) == 1 {
```

Comment:
using valueDigest instead of value, because that is the only one set when doing proof validation